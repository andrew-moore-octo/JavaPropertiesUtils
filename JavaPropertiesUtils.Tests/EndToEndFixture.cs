using System;
using Assent;
using NUnit.Framework;

namespace JavaPropertiesUtils.Tests
{
    public class EndToEndFixture
    {
        [Test]
        public void Test()
        {
            var input = ResourceUtils.ReadEmbeddedResource("multiple-pairs-and-comments.properties");
            var parsed = Parser.Parse(input)
                .Set("key.2", "The new value for key 2")
                .Set("key=4", "The new value for key 4")
                .Set("Non existent key", "Should not be found.");

            var actual = parsed.ToString();
            this.Assent(actual);
        }
    }
}