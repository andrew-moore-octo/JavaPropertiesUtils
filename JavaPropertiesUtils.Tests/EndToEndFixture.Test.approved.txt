# Test file
KEY1 = value1

! Another comment
   key.2: The new value for key 2
key\ 3 : third value
key\=4=     The new value for key 4
key\\\ \:\==fifth value
